index.js

```diff
@@ -0,0 +1,2 @@
isionDelay=0,this.addBlocksToStackedArray(this.controlBlock.blockArray,this.stackedBlock.blockArray),!function(t){t>0&&(G+=t*t,q.textContent=`${G}`)}(this.removeCompletedLine(0)),this.checkIsGameOver(this.stackedBlock.blockArray)?(this.controlBlock.removeControlBlock(),this.audioPlayer.play(P),X()):(this.controlBlock.addNewControlBlock(),F++,K.textContent=`${F}`,this.flowGravityWithDraw(),U())}}removeCompletedLine(t){let e=this.stackedBlock.blockArray;for(let i=0;i<28;i++){let o=!0;for(let t=3;t<13;t++)if(!e[t][i].isExist){o=!1;break}if(o)return this.removeLine(i),this.removeCompletedLine(t+1)}return t>0&&this.audioPlayer.play(D),t}removeLine(t){let e=this.stackedBlock.blockArray;for(let i=t;i>0;i--)for(let t=0;t<16;t++)A(e[t][i],e[t][i-1])}addBlocksToStackedArray(t,e){for(let i=0;i<16;i++)for(let o=0;o<28;o++)t[i][o].isExist&&A(e[i][o],t[i][o]);this.shouldUpdateStackedLayer=!0}shiftControlBlock(){if(!this.shiftBlock.isAlreadyShiftedThisTime){if(this.shiftBlock.isAlreadyShiftedThisTime=!0,this.audioPlayer.play(M),this.shiftBlock.isShiftedBlockEmpty())this.shiftBlock.setShiftBlock(this.controlBlock.controlBlockType),this.controlBlock.addNewControlBlock();else{let t=this.shiftBlock.shiftedBlock;this.shiftBlock.setShiftBlock(this.controlBlock.controlBlockType),this.changeControlBlock(t)}this.flowGravityWithDraw()}}changeControlBlock(t){this.controlBlock.changeControlBlock(t)}checkIsGameOver(t){for(let e=3;e<13;e++)if(t[e][1].isExist)return!0;return!1}reDraw(){this.drawBlocks()}onEventLeftArrow(){w(this.controlBlock.blockArray,this.stackedBlock.blockArray)&&f(this.controlBlock.blockArray)}onEventRightArrow(){g(this.controlBlock.blockArray,this.stackedBlock.blockArray)&&y(this.controlBlock.blockArray)}onEventDownArrow(){this.flowGravityWithDraw()}onEventUpArrow(){this.controlBlock.rotate(this.controlBlock,this.controlBlock.blockArray,this.stackedBlock.blockArray,3,!0)}onEventSpace(){for(this.isSpaceDownRunning=!0;this.isSpaceDownRunning;)this.flowGravity();this.drawBlocks()}onEventShift(){this.shiftControlBlock()}onEventKeyQ(){this.controlBlock.rotate(this.controlBlock,this.controlBlock.blockArray,this.stackedBlock.blockArray,3,!1)}onEventKeyE(){this.controlBlock.rotate(this.controlBlock,this.controlBlock.blockArray,this.stackedBlock.blockArray,3,!0)}onEventKeyZ(){this.controlBlock.rotate(this.controlBlock,this.controlBlock.blockArray,this.stackedBlock.blockArray,3,!1)}onEventKeyC(){this.controlBlock.rotate(this.controlBlock,this.controlBlock.blockArray,this.stackedBlock.blockArray,3,!0)}};window.onkeydown=t=>{!function(t){if(t.preventDefault(),H)return;switch(t.code){case"ArrowLeft":Y.onEventLeftArrow();break;case"ArrowRight":Y.onEventRightArrow();break;case"ArrowDown":Y.onEventDownArrow();break;case"ArrowUp":Y.onEventUpArrow();break;case"Space":Y.onEventSpace();break;case"ShiftLeft":Y.onEventShift();break;case"KeyQ":Y.onEventKeyQ();break;case"KeyE":Y.onEventKeyE();break;case"KeyZ":Y.onEventKeyZ();break;case"KeyC":Y.onEventKeyC()}Y.drawBlocks()}(t)};
```

Comment:
My gut feeling is this file should be git ignored because it is generated file from bundler with `"parcel build src/index.js --dist-dir dist/client",`. 
Maybe some github action could support automatically build and deploy to the repository. 
The manual build (like, build and then commit) could increase error possibility.

WDYT?

Reply:
IDTS actually, it's a essential file to host [git pages](https://pages.github.com/).

the "git pages" hosts **index.html** by default, and **index.html** refers to this compressed **index.js**.
that's why index.js should be remain.

Reply:
Yep, this file is required. but I think not a `master` or `develop` branch. 
I guess currently deployment system use `develop`  branch for github-page. 
And I think we should use  `gh-page` branch if we need to build with bundle or webpack.
So we might be needed github-action to automatically build from the `master` branch when it is updated to gh-page branch.

BTW, If we should keep this file on dev or master, Can I ask your help me understand PR flow? 

My expectation of current:
1. build with bundle
2. git add updated files included index.js
3. git commit
4. git push

My concerns:
1. If I forgat build, what will happen? 
2. What if happened merge conflict with index.js by different PRs?

Reply:
I think @WoochanLee should find GH-action for the above working. As I know, the only owner could set the gh-action.

Reply:
I got ur concerns. but having different .ignore files between gh-page and others seems more inconsistency.

Reply:
I think Build and deploy might not need separate .gitignore. 

I found here [1] to command for force-add.



The expectation flow:
1. git add
2. git commit
3. git push
4. approved by reviewers
5. merged to develop branch
6. deploy request by the owner to GitHub action (maybe the current develop branch will be merged to master). 
7. Github action will get master and build.
8. [in github action] add -f index.js 
9. [in github action] push -f origin gh-page
10. Deployment is completed.

[1] https://stackoverflow.com/questions/8006393/force-add-despite-the-gitignore-file

Reply:
I will check github action. May need some time.